modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp

```diff
@@ -51,6 +51,33 @@ std::string LemDCBM400600Controller::get_current_transaction() {
     }
 }
 
+void LemDCBM400600Controller::update_lem_status() {
+    // should call this after a communication error to figure out what has been happening
+    auto status_response = this->http_client->get("/v1/status");
+
+    if (status_response.status_code != 200) {
+        throw UnexpectedDCBMResponseCode("/v1/status", 200, status_response);
+    }
+    try {
+        json data = json::parse(status_response.body);
+        this->transaction_is_ongoing_at_startup = data.at("status").at("bits").at("transactionIsOnGoing");
+        this->current_transaction_id = get_current_transaction();
+        if (this->transaction_is_ongoing_at_startup) {
+            // we need to get the current transaction id or the last known transaction id since we might
+            // receive a stop transaction with id 0 or with the last known transaction if
+            // meaning that the system had a failure and the transaction was started but id was not saved
+            // and we try to recover from the error, thus we need to cancel the transaction
+            EVLOG_warning << "LEM DCBM 400/600: A transaction is already ongoing and it has the id:"
+                          << this->current_transaction_id;
+        } else {
+            EVLOG_info << "LEM DCBM 400/600: The last known transaction has the id:" << this->current_transaction_id;
+        }
+    } catch (json::exception& json_error) {
```

Comment:
for readability, the json parsing could be moved into its own helper function like:
```
auto get_transaction_ongoing_at_startup(const StatusResponseType& response) {
  try {
    const auto& data = json::parse(response.body);
    return data.at("status").at("bits").at("transactionIsOnGoing");
  } catch (const json::exception& json_error) {
    throw UnexpectedDCBMResponseBody(
            "/v1/status", fmt::format("Json error {} for body {}", json_error.what(), status_response.body));
  }
}
```

Reply:
then the main code would just look like
```
this->transaction_is_ongoing_at_startup = get_transaction_ongoing_at_startup(status_response);
if (this->transaction_is_ongoing_at_startup) {
  EVLOG_warning ....
} else {
  EVLOG_info ....
}
this->current_transaction_id = get_current_transaction();
```

Reply:
much more readable in *my* point of view